pkg/controller/grouping/controller.go

```diff
@@ -71,6 +73,21 @@ func ExternalEntityIPsIndexFunc(obj interface{}) ([]string, error) {
 	return nil, nil
 }
 
+func NodeIPsIndexFunc(obj interface{}) ([]string, error) {
+	node, ok := obj.(*v1.Node)
+	if !ok {
+		return nil, fmt.Errorf("obj is not node: %+v", obj)
```

Comment:
Since you're checking if obj is of type *v1.Node. However, when the type assertion fails, consider providing more context in the error message. For example: "Expected *v1.Node, got %T"